src/client/datascience/interactive-common/interactiveBase.ts

```diff
@@ -911,7 +911,7 @@ export abstract class InteractiveBase extends WebviewPanelHost<IInteractiveWindo
 
         if (providerConnection) {
             try {
-                await this.ensureNotebook(providerConnection);
+                await this.ensureNotebook(providerConnection, true);
```

Comment:
Not sure why this one is true and the other one is false?

Reply:
I found it by testing

Reply:
That makes it hard to reason about this later. Is there a more concrete way to follow this up to the decision about using auto start or not?

Reply:
A clue I got was that since this function already hardcodes `disableUI` to false, that one call should also hardcode a false (it makes sense this is called when running a cell). And there's only one more place where `ensureNotebook` is called.

![image](https://user-images.githubusercontent.com/20347396/113943838-53c30b00-97b8-11eb-9c27-b299d17d7599.png)

Reply:
Maybe some comments would help?

Reply:
In your picture I'm not sure that disableUI should be false or not. Because I don't know how that decision was made.

That also means on the spot where you have true, you should also have disableUI: true for ensureConnection.

I think it would just be clearer if the disableUI flag was pushed even higher. Like createNotebookIfProviderConnectionExists is called in the ctor, that one seems obvious that you'd want to disable the UI. The other spot where's it called, maybe not?

Reply:
Could maybe make the default to false for the disableUI parameter? It seems like it's the exceptional case the disable the UI. So you could call the true out here with a comment as to why, and the other spots don't need a random false drifting around.

@rchiodo would that be cleaner?

Reply:
I'd still want to know why it's true at this spot. I think the true spot isn't always correct either. Based on how ```createNotebookIfProviderConnectionExists``` is called. It should be true on the first call (in the ctor). But should it be for notebookProvider.connectionMade? Only if notebookProvider was called with disableUI set to true itself.